### PR TITLE
Do not render widget when no token config found 

### PR DESCRIPTION
**Pull request by emreboga**
This is a very unlikely case unless a user tempers with the localStorage entry. It was discovered while testing Portal with an invalid localStorage entry. To fix this potential problem we are adding another layer of verification to filter out the entries with no token configs.

### CODE REVIEW

File: wormhole-connect/src/views/v2/Bridge/ReviewTransaction/GasSlider.tsx [hunk 1]
```diff
@@ -117,7 +117,6 @@ const GasSlider = (props: {
     const tokenAmount = amount.display(
       amount.truncate(props.destinationGasDrop, 6),
     );
-    console.log(tokenAmount);
```
Comment (emreboga) [inline wormhole-connect/src/views/v2/Bridge/ReviewTransaction/GasSlider.tsx]: This one was creating some noise in the console when in Review Transaction view. Please let me know if you think it'd still be useful when debugging.

File: wormhole-connect/src/views/v2/TxHistory/Widget/Item.tsx [hunk 2]
```diff
@@ -108,7 +108,8 @@ const WidgetItem = (props: Props) => {
     toChain,
     token: tokenTuple,
   } = txDetails || {};
-  const token = config.tokens.get(tokenTuple);
+
+  const tokenConfig = config.tokens.get(tokenTuple);
```
Comment (artursapek) [inline wormhole-connect/src/views/v2/TxHistory/Widget/Item.tsx]: can we stop using the term "token config"? just keep this as-is.

File: wormhole-connect/src/views/v2/TxHistory/Widget/index.tsx [hunk 3]
```diff
@@ -43,7 +44,25 @@ const TxHistoryWidget = () => {
 
   useEffect(() => {
     // Get all in-progress transactions from localStorage
-    setTransactions(getTxsFromLocalStorage());
+    const txs = getTxsFromLocalStorage();
+
+    // Filter out the ones with unknown token configs
```
Comment (artursapek) [inline wormhole-connect/src/views/v2/TxHistory/Widget/index.tsx]: again, please don't call them token configs. they're just tokens.

File: wormhole-connect/src/views/v2/TxHistory/Widget/index.tsx [hunk 4]
```diff
@@ -43,7 +44,25 @@ const TxHistoryWidget = () => {
...
+    const verifiedTxs = txs?.filter((tx) => {
+      if (!tx?.txDetails?.token) {
+        return false;
+      }
+      try {
+        return !!config.tokens.get(tx?.txDetails?.token);
```
Comment (artursapek) [inline wormhole-connect/src/views/v2/TxHistory/Widget/index.tsx:55]: There's a better way to handle this edge case. If there's a pending tx, then I think we can safely assume the tokens it refers to exist. And if that's true, and they are not in the token cache, you should just `getOrFetchToken` them here. You can ignore the result - just kick it off and the `TokensContext` will fetch the token and put it in the cache in the background. [See this for example](https://github.com/wormhole-foundation/wormhole-connect/blob/development/wormhole-connect/src/views/v2/Bridge/AssetPicker/TokenList.tsx#L77).

This way you don't have to filter this thing out, you can render it properly once that fetch is complete.

You'll probably need to handle the "loading" case in the meantime where the token is not yet in the cache... maybe just render a `Skeleton`. And re-render once the cache is updated (which you can do by depending on `lastTokenCacheUpdate` [like here](https://github.com/wormhole-foundation/wormhole-connect/blob/development/wormhole-connect/src/views/v2/Bridge/index.tsx#L323))

**Review by artursapek — CHANGES_REQUESTED** [review]

File: wormhole-connect/src/views/v2/TxHistory/Widget/index.tsx (hunk 4 above)
Comment (emreboga) [reply wormhole-connect/src/views/v2/TxHistory/Widget/index.tsx:55]: Thanks for the recommendation, but it has the assumption that the token is among the supported ones. The need for this check came in after I learnt that Portal QA team has a test case to make sure Connect is not broken after user tempers with the local storage entry. One of them was the token. In that case it doesn't matter whether we fetch the token. This is still a very edge case and I'm ok leaving it out if you think it's unnecessary.

File: wormhole-connect/src/views/v2/TxHistory/Widget/Item.tsx (hunk 2 above)
Comment (emreboga) [reply wormhole-connect/src/views/v2/TxHistory/Widget/Item.tsx]: https://github.com/wormhole-foundation/wormhole-connect/pull/3056/commits/e84f5c3817342ff641d038b9f3934c940db55db4

File: wormhole-connect/src/views/v2/TxHistory/Widget/index.tsx (hunk 3 above)
Comment (emreboga) [reply wormhole-connect/src/views/v2/TxHistory/Widget/index.tsx]: https://github.com/wormhole-foundation/wormhole-connect/pull/3056/commits/e84f5c3817342ff641d038b9f3934c940db55db4

File: wormhole-connect/src/views/v2/TxHistory/Widget/index.tsx (hunk 4 above)
Comment (artursapek) [reply wormhole-connect/src/views/v2/TxHistory/Widget/index.tsx:55]: OK. Fine with me. Unlikely the non-tamper case would ever happen to begin with anyway

**Review by artursapek — APPROVED** [review]
